src/azure-cli-core/azure/cli/core/profiles/shared.py

```diff
@@ -0,0 +1,158 @@
+    DATA_STORAGE = ('azure.multiapi.storage', None)
+
+    def __init__(self, import_prefix, client_path):
+        """Constructor.
+
+        :param import_prefix: Path to the (unversioned) module.
+        :type import_prefix: str.
+        :param client_path: The path to the client for this resource type.
+        :type client_path: str.
+        """
+        self.import_prefix = import_prefix
+        self.client_path = client_path
+
+
+AZURE_API_PROFILES = {
+    'latest': {
+        ResourceType.MGMT_STORAGE: '2016-12-01',
+        ResourceType.MGMT_NETWORK: '2016-09-01',
+        ResourceType.MGMT_CONTAINER_SERVICE: '2017-01-31',
+        ResourceType.MGMT_COMPUTE: '2016-04-30-preview',
+        ResourceType.MGMT_RESOURCE_FEATURES: '2015-12-01',
+        ResourceType.MGMT_RESOURCE_LINKS: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_LOCKS: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_POLICY: '2016-04-01',
+        ResourceType.MGMT_RESOURCE_RESOURCES: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS: '2016-06-01',
+        ResourceType.DATA_STORAGE: '2016-05-31'
+    },
+    '2016-sample': {
+        ResourceType.MGMT_STORAGE: '2016-12-01',
+        ResourceType.MGMT_NETWORK: '2016-09-01',
+        ResourceType.MGMT_CONTAINER_SERVICE: '2017-01-31',
+        ResourceType.MGMT_COMPUTE: '2016-04-30-preview',
+        ResourceType.MGMT_RESOURCE_FEATURES: '2015-12-01',
+        ResourceType.MGMT_RESOURCE_LINKS: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_LOCKS: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_POLICY: '2016-12-01',
+        ResourceType.MGMT_RESOURCE_RESOURCES: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS: '2016-06-01',
+        ResourceType.DATA_STORAGE: '2016-05-31'
+    },
+    '2015-sample': {
+        ResourceType.MGMT_STORAGE: '2015-06-15',
+        ResourceType.MGMT_NETWORK: '2015-06-15',
+        ResourceType.MGMT_CONTAINER_SERVICE: '2017-01-31',
+        ResourceType.MGMT_COMPUTE: '2015-06-15',
+        ResourceType.MGMT_RESOURCE_FEATURES: '2015-12-01',
+        ResourceType.MGMT_RESOURCE_LINKS: '2016-09-01',
+        ResourceType.MGMT_RESOURCE_LOCKS: '2015-01-01',
+        ResourceType.MGMT_RESOURCE_POLICY: '2016-04-01',
+        ResourceType.MGMT_RESOURCE_RESOURCES: '2016-02-01',
+        ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS: '2016-06-01',
+        ResourceType.DATA_STORAGE: '2015-04-05'
+    }
+}
+
+
+def get_api_version(api_profile, resource_type):
+    """Get the API version of a resource type given an API profile.
+
+    :param api_profile: The name of the API profile.
+    :type api_profile: str.
+    :param resource_type: The resource type.
+    :type resource_type: ResourceType.
+    :returns:  str -- the API version.
+    :raises: APIVersionException
+    """
+    try:
+        return AZURE_API_PROFILES[api_profile][resource_type]
+    except KeyError:
+        raise APIVersionException(resource_type, api_profile)
+
+
+def get_client_class(resource_type):
```

Comment:
This whole `shared.py` file should not be used by anyone else.
`azure/cli/core/profiles/__init__.py` is what will be used by others and that has the following public methods: `get_api_version`, `get_sdk`, `get_versioned_sdk_path` each of which has their use.

I have renamed `shared.py` to `_shared.py` to make this clearer.